app/RouteHandlers/AddDonationHandler.php

```diff
@@ -28,36 +34,62 @@ public function __construct( FunFunFactory $ffFactory, Application $app ) {
 		$this->app = $app;
 	}
 
-	public function handle( Request $request ) {
+	public function handle( Request $request ): Response {
 		$responseModel = $this->ffFactory->newAddDonationUseCase()->addDonation(
 			$this->createDonationRequest( $request )
 		);
 
-		if ( $responseModel->isSuccessful() ) {
-			$donation = $responseModel->getDonation();
-
-			switch( $donation->getPaymentType() ) {
-				case PaymentType::DIRECT_DEBIT:
-				case PaymentType::BANK_TRANSFER:
-					return $this->ffFactory->newAddDonationHtmlPresenter()->present( $donation );
-				case PaymentType::PAYPAL:
-					return $this->app->redirect(
-						$this->ffFactory->newPayPalUrlGenerator()->generateUrl(
-							$donation->getId(),
-							$donation->getAmount(),
-							$donation->getInterval(),
-							'token',
-							'utoken'
-						)
-					);
-				case PaymentType::CREDIT_CARD:
-					return $this->ffFactory->newCreditCardPaymentHtmlPresenter()->present( $responseModel );
-			}
-			// TODO: take over confirmation page selection functionality from old application
-			// TODO: return update token
+		if ( !$responseModel->isSuccessful() ) {
+			return new Response( 'TODO: error occurred' ); // TODO
+		}
+
+		// TODO: take over confirmation page selection functionality from old application
```

Comment:
That might be, I'm not sure. I did not change the functionality of that code (just refactored it a bit). Same with the TODO, I just moved it. Perhaps Kai finished the TODO and did not remove it yet?